src/Controller/Api/QuapController.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace App\Controller\Api;
+
+use App\DTO\Model\WidgetControllerData\WidgetRequestData;
+use App\Entity\Group;
+use App\Exception\ApiException;
+use App\Service\QuapService;
+use DateTime;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\ParamConverter;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Exception\BadRequestException;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+
+class QuapController extends AbstractController
+{
+    public function getQuestionnaireData(
+        Request     $request,
+        QuapService $quapService,
+        string      $type
+    ): JsonResponse
+    {
+        $date = $request->get('date', null);
+
+        $date = $date ? DateTime::createFromFormat('Y-m-d', $date) : new DateTime("now");
+
+        return $this->json($quapService->getQuestionnaireDataByType($type, $request->getLocale(), $date));
```

Comment:
The Controller (in Symfony) has 4 Tasks.

1. Get the data for the request
2. Validate the data
3. Call a service to handle the data
4. Prepare the response from the data returned by the service

You are missing the 4. Step here. Never just return the Entity from the DB. You may return data that you really don't want to return. (Security / Privacy concerns) Always create a DTO where you control what's being returned yourself.